codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/localservice/nameresolver/DafnyNameResolver.java

```diff
@@ -196,4 +215,17 @@ public static String getDafnyDependentErrorType(
       .concat(DOT)
       .concat(sdkId);
   }
+
+  public static String getDafnyUnionBaseStructType(
+    final Shape shape,
+    final String memberName
+  ) {
+    return DafnyNameResolver
+      .getDafnyCompanionStructType(shape, memberName)
+      .concat("{}");
+  }
+
+  public static String dafnyCompilesExtra_(final String name) {
```

Comment:
Is there any way to just reuse this implementation? https://github.com/smithy-lang/smithy-dafny/blob/180ad67ad9f80a8b5f9272b06dbbdcce24d4f369/codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/utils/DafnyNameResolverHelpers.java#L46

Reply:
Let me do this once the code base has stabilized, during the final review before main merge.